### PR TITLE
2671372 location as entity

### PR DESCRIPTION
**Pull request by olafkarsten**
Replaces [PR9](https://github.com/BBGuy/commerce_stock/pull/9) and removes non essential changes as requested by @BBGuy

### CODE REVIEW

File: src/StockAvailabilityChecker.php [hunk 1]
```diff
@@ -51,15 +51,17 @@ public function check(PurchasableEntityInterface $entity, $quantity, Context $co
     }
```
Comment (BBGuy) [inline src/StockAvailabilityChecker.php:51]: I don't see what changes are relevant in this file?

File: modules/local_storage/src/LocalStockChecker.php [hunk 2]
```diff
@@ -192,23 +205,21 @@ public function getIsStockManaged(PurchasableEntityInterface $entity) {
    * {@inheritdoc}
    */
   public function getLocationList($return_active_only = TRUE) {
```
Comment (BBGuy) [inline modules/local_storage/src/LocalStockChecker.php:207]: We are removing the active part of the array but that's OK as the entity holds its status. 
Nice rewrite.

File: src/StockAvailabilityChecker.php (hunk 1 above)
Comment (olafkarsten) [reply src/StockAvailabilityChecker.php:51]: Microperformanceoptimization. If we we need to return, do it before we request stockConfig and fetch the locations

File: src/StockLocationInterface.php [hunk 3]
```diff
@@ -0,0 +1,37 @@
+<?php
```
Comment (BBGuy) [inline src/StockLocationInterface.php:1]: This is change needs highlighting as it shows how we are not only adding location as entities in the local stock implementation but making the stock manager aware of those using this interface. Great work

File: src/StockAvailabilityChecker.php (hunk 1 above)
Comment (BBGuy) [reply src/StockAvailabilityChecker.php:51]: ok I can see that but lets push it in a cleanup/general commit. I want the commit to show what needed changing for this to work. 
For me its meaningful to be able to easily tell that the location as entity did not change the StockAvailabilityChecker.  
No need to update the PR. will do it in the commit.

**Comment by BBGuy** [comment]
Reviewed all the code and it looks great. will do some testing and commit

**Comment by BBGuy** [comment]
Supper great work stock locations are now entities. 
The commit with the location changes:
https://github.com/BBGuy/commerce_stock/commit/4f1539487a917fb06a0f22dc104d61535ddcd080
I have separated some of the code cleanup changes that did not directly relate to this task and push those as https://github.com/BBGuy/commerce_stock/commit/fac86adb269fe09cf2e497d3df55edc707b221bb

**Comment by steveoliver** [comment]
This is not done. For one, there is no 'location_id' field to use as the Location entity 'id' -- https://github.com/BBGuy/commerce_stock/pull/25/files#diff-8791b543d4d923f3589357089923deecR47

**Comment by steveoliver** [comment]
Well I spoke too soon -- the 'location_id' does not need to exist as a field, it will be used as the name of the 'id' column when the entity type table is created.

I noticed this PR said there were unmerged commits after the PR was last closed.  @olafkarsten - What might those be?

Also, is there any reason this PR should not have [had] a passing build before commit?

**Comment by BBGuy** [comment]
@steveoliver looked to me like the failed builds did not relate to changes made by this PR and I wanted us to move ahead.  
Re changes @olafkarsten can provide more details but I asked him to remove changes not directly related to this issue. I also separated some of his work into a separate commit as it was general code cleanup not relating to the issue. see [above](https://github.com/BBGuy/commerce_stock/pull/25/#issuecomment-284221849)

**Comment by BBGuy** [comment]
Will pay more attention to the travis build once we get it working again 👍

**Comment by BBGuy** [comment]
This is now live so closing, if any code was missed out then lets open a new PR

**Pull request closed without merging**